File: data/readable_format/cpp/file734.cpp
#include <iostream>

// Assuming the structure of sph_boundary_type based on the original context
struct sph_boundary_type {
    int dummy; // Placeholder member variable. Replace with actual structure.
};

class MBoundaryParamsSphMHD {
public:
    sph_boundary_type sph_bc_U;
    sph_boundary_type sph_bc_B;
    sph_boundary_type sph_bc_T;
    sph_boundary_type sph_bc_C;

    double* vp_ICB_bc = nullptr;
    double* vt_ICB_bc = nullptr;
    double* vp_CMB_bc = nullptr;
    double* vt_CMB_bc = nullptr;

    void allocate_vsp_bc_array(int jmax) {
        vp_ICB_bc = new double[jmax]{};
        vt_ICB_bc = new double[jmax]{};
        vp_CMB_bc = new double[jmax]{};
        vt_CMB_bc = new double[jmax]{};
    }

    void deallocate_vsp_bc_array() {
        delete[] vp_ICB_bc;
        vp_ICB_bc = nullptr;
        delete[] vt_ICB_bc;
        vt_ICB_bc = nullptr;
        delete[] vp_CMB_bc;
        vp_CMB_bc = nullptr;
        delete[] vt_CMB_bc;
        vt_CMB_bc = nullptr;
    }

    ~MBoundaryParamsSphMHD() {
        deallocate_vsp_bc_array();
    }
};

void testMBoundaryParamsSphMHD() {
    const int jmax = 5;
    MBoundaryParamsSphMHD obj;

    // Allocate the arrays
    obj.allocate_vsp_bc_array(jmax);

    // Check initialization
    for (int i = 0; i < jmax; ++i) {
        if (obj.vp_ICB_bc[i] != 0.0 || obj.vt_ICB_bc[i] != 0.0 ||
            obj.vp_CMB_bc[i] != 0.0 || obj.vt_CMB_bc[i] != 0.0) {
            std::cerr << "Error: Arrays not properly initialized." << std::endl;
            return;
        }
    }
    std::cout << "Allocation and initialization test passed." << std::endl;

    // Deallocation is handled by the destructor
    std::cout << "Deallocation test passed." << std::endl;
}

int main() {
    testMBoundaryParamsSphMHD();
    return 0;
}

File: data/readable_format/fortran/file734.f90
program test_m_boundary_params_sph_MHD
  implicit none

  ! Type definitions
  type sph_boundary_type
    ! Assuming the structure of sph_boundary_type. Adjust according to actual definition.
    integer :: dummy
  end type sph_boundary_type

  ! Declarations similar to those in the module m_boundary_params_sph_MHD
  type(sph_boundary_type) :: sph_bc_U
  type(sph_boundary_type) :: sph_bc_B
  type(sph_boundary_type) :: sph_bc_T
  type(sph_boundary_type) :: sph_bc_C

  real, allocatable :: vp_ICB_bc(:)
  real, allocatable :: vt_ICB_bc(:)
  real, allocatable :: vp_CMB_bc(:)
  real, allocatable :: vt_CMB_bc(:)

  integer, parameter :: jmax = 5
  integer :: i

  ! Allocate the arrays
  call allocate_vsp_bc_array(jmax)

  ! Check initialization
  do i = 1, jmax
    if (vp_ICB_bc(i) /= 0.0 .or. vt_ICB_bc(i) /= 0.0 .or. &
        vp_CMB_bc(i) /= 0.0 .or. vt_CMB_bc(i) /= 0.0) then
      print *, "Error: Arrays not properly initialized."
      stop
    endif
  end do
  print *, "Allocation and initialization test passed."

  ! Deallocate the arrays
  call deallocate_vsp_bc_array
  print *, "Deallocation test passed."

contains

  subroutine allocate_vsp_bc_array(jmax)
    integer, intent(in) :: jmax

    allocate(vp_ICB_bc(jmax))
    allocate(vt_ICB_bc(jmax))
    allocate(vp_CMB_bc(jmax))
    allocate(vt_CMB_bc(jmax))
    vp_ICB_bc = 0.0
    vt_ICB_bc = 0.0
    vp_CMB_bc = 0.0
    vt_CMB_bc = 0.0
  end subroutine allocate_vsp_bc_array

  subroutine deallocate_vsp_bc_array
    deallocate(vp_ICB_bc, vt_ICB_bc)
    deallocate(vp_CMB_bc, vt_CMB_bc)
  end subroutine deallocate_vsp_bc_array

end program test_m_boundary_params_sph_MHD